### DISCUSSION TITLE
Hue Lights Color won't change

### DISCUSSION THREAD

**Discussion by bigtcze** [Koenkk/zigbee2mqtt#8948] [Devices]
Hello guys, could you please help me? I got couple Hue Lights "Philips 9290022169" at home connected to Zigbee2MQTT obviously. When I try to change color for them, it works only when light's brightness is bigger than 0. Do you know how to change their color, when brightness is 0? I tried to set "color_temp_startup" along with usual "color_temp". I analysed logs and lights are accepting new color value without issue:

info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/color_mode', payload 'color_temp'
info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/update-state', payload 'idle'
info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/state', payload 'OFF'
info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/brightness', payload '0'
info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/color_temp', payload '450'
info  2021-10-01 21:27:03: MQTT publish: topic 'zigbee2mqtt/friendly_name/linkquality', payload '90'

but when I change brightness, their color is still on the previous color, when they had brightness > 0. 

Description of my issue:
1) Light have brightness 50 (something > 0)
2) I change color to 200
3) I change brightness to 0
4) Then I change color to 400
5) and then set brightness back to 200 and color is still the same as in the point 2 and light is reporting color 400 to logs

**Comment by mkaut** [comment]
I experience the same issue, with the same Hue light, but I thought to post my actual logs, in case they help someone to trace the issue.
Let's say I have the light on with color_temp=155 (cold light) and switch it off (change state to 'off').

Then I want to change it to warm, so I set color_temp to 450 and, just to test the effect, color_temp_startup to 400:
```bash
mosquitto_pub -h 192.168.1.16 -t 'zigbee2mqtt/philips_hue_wa_1/set/color_temp' -m '450'
mosquitto_pub -h 192.168.1.16 -t 'zigbee2mqtt/philips_hue_wa_1/set/color_temp_startup' -m '400'
```
This results in the following messages (captured by mosquitto_sub):
```bash
zigbee2mqtt/philips_hue_wa_1/set/color_temp 450
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/color_temp 155
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/state OFF
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/state OFF
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 153
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:22:52+02:00
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 153
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:22:52+02:00
zigbee2mqtt/philips_hue_wa_1/set/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/state OFF
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/state OFF
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 153
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:23:09+02:00
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:23:09+02:00
```
This suggests that both values have been changed. It also shows that the light (or z2m?) reports first the old value and then the new one.

Now, when I change the state back to ON, I get
```
zigbee2mqtt/philips_hue_wa_1/set/state ON
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/state OFF
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/state ON
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:23:40+02:00
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:23:40+02:00
```
so the light seems to have kept the new value, but this is __wrong__ as the light shines with the same cold light it had before switching off.
Indeed, running `mosquitto_pub -h 192.168.1.16 -t 'zigbee2mqtt/philips_hue_wa_1/get' -m '{"color_temp": ""}'` returns
```
zigbee2mqtt/philips_hue_wa_1/get {"color_temp": ""}
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/brightness 231
zigbee2mqtt/philips_hue_wa_1/color_temp 450
zigbee2mqtt/philips_hue_wa_1/color_temp 155
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/linkquality 63
zigbee2mqtt/philips_hue_wa_1/state ON
zigbee2mqtt/philips_hue_wa_1/state ON
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/update-state idle
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/update_available false
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/color_mode color_temp
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/color_temp_startup 400
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:26:19+02:00
zigbee2mqtt/philips_hue_wa_1/last_seen 2021-10-10T21:26:19+02:00
```
showing the change of color_temp from 450 back to 155.

What is going on? And is it expected, or is this a but which should be reported?

Comment: what I _actually_ want to do is to be able to change the color temperature during the day. I was hoping to be able to send the color_temp updates to all lights at given time points, but that works only if they accept the change also when they are off...

**Comment by PaulHamdinger** [comment]
A possible solution (for some bulbs) was discussed here : https://github.com/Koenkk/zigbee2mqtt/issues/13562

@sjorge mentions that enabling the "execute_if_off" flag of "color_options" if you have certain Hue or Trådfri bulbs : https://github.com/Koenkk/zigbee2mqtt/issues/13562#issuecomment-1217712911

This works for my Hue GU10 Color and White Ambiance bulb (newer, Bluetooth enabled). But fails for my White Ambiance (only) canister lights and bulbs.

If you'd like to experiment, you can enable the flag like so, if you have a Mosquitto client to play with...

`mosquitto_pub -t 'zigbee2mqtt/ikea-desk-lamp/set/color_options' -m '{"execute_if_off":true}' -q 2`
